core/arch/arm/kernel/thread.c

```diff
@@ -47,7 +47,7 @@
 #ifdef CFG_CORE_SANITIZE_KADDRESS
 #define STACK_TMP_SIZE		(3072 + STACK_TMP_OFFS)
 #else
-#define STACK_TMP_SIZE		(1536 + STACK_TMP_OFFS)
+#define STACK_TMP_SIZE		(2048 + STACK_TMP_OFFS)
```

Comment:
With Clang, more stack space will be consumed?

Reply:
Yes, it looks like so. Increasing stack size fixed a couple of problems in my tests, including "dead canaries" on boot.